### PR TITLE
Update theme and intersphinx

### PR DESCRIPTION
**Pull request by lilyminium**

### CODE REVIEW

File: doc/source/conf.py [hunk 1]
```diff
@@ -32,120 +31,106 @@
...
-           u'Dominik Mierzejewski, Giacomo Fiorin, Lily Wang, Josh Vermaas, '
-           u'Irfan Alibay，Zhiyi Wu')
-copyright = u'2007-2021, ' + authors
+project = "GridDataFormats"
+authors = (
+    "Oliver Beckstein, Jan Domanski, Jesse Johnson, Max Linke, Tyler Luchko, "
+    "Dominik Mierzejewski, Giacomo Fiorin, Lily Wang, Josh Vermaas, "
+    "Irfan Alibay，Zhiyi Wu"
+)
+copyright = "2007-2021, " + authors
```
Comment (IAlibay) [inline doc/source/conf.py:72]:
```suggestion
copyright = "2007-2023, " + authors
```

File: environment.yaml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+name: griddataformats-docs
+channels:
+  - conda-forge
+
+dependencies:
+  - python
+  - pip
```
Comment (IAlibay) [inline environment.yaml:7]:
```suggestion
  - pip
  # base deps
  - numpy
  - scipy
  - mrcfile
```

**Comment by orbeckst** [comment]
I'll have a look if I can get it to work as is.

File: environment.yaml [hunk 3]
```diff
@@ -0,0 +1,13 @@
+name: griddataformats-docs
```
Comment (IAlibay) [inline environment.yaml:1]: This should just be merged into what's in `ci/environment.yaml`

**Review by IAlibay — COMMENTED** [review]
@orbeckst I think this will now pick up the wrong yaml file

File: environment.yaml (hunk 3 above)
Comment (orbeckst) [reply environment.yaml:1]: I know. Just figured it out locally.

**Comment by orbeckst** [comment]
Locally I can't get the sphinx stuff to install in a Python 3.12 environment:
```
Pinned packages:
  - python 3.12.*


Could not solve for environment specs
The following package could not be installed
└─ mdanalysis-sphinx-theme >=1.0.1  is installable and it requires
   └─ sphinx >=6.2.1,<7.0 , which requires
      └─ docutils >=0.18.1,<0.20  with the potential options
         ├─ docutils [0.18.1|0.19] would require
         │  └─ python >=3.10,<3.11.0a0 , which can be installed;
         ├─ docutils [0.18.1|0.19] would require
         │  └─ python >=3.7,<3.8.0a0 , which can be installed;
         ├─ docutils [0.18.1|0.19] would require
         │  └─ python >=3.8,<3.9.0a0 , which can be installed;
         ├─ docutils [0.18.1|0.19] would require
         │  └─ python >=3.9,<3.10.0a0 , which can be installed;
         └─ docutils [0.18.1|0.19] would require
            └─ python >=3.11,<3.12.0a0 , which can be installed.
```
so need to do sphinx in 3.10 or so.

**Comment by orbeckst** [comment]
ok, need to pip-install sphinx_sitemap — somehow I had hoped that this was magically included in mdanalysis-sphinx-theme (but on reflection it's sensible that it isn't).
```
 Could not import extension sphinx_sitemap (exception: No module named 'sphinx_sitemap')
```

**Review by orbeckst — APPROVED** [review]

**Comment by orbeckst** [comment]
Thank you @lilyminium and also thanks @IAlibay !

**Pull request merged**